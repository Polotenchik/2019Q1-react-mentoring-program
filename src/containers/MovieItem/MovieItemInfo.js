import React from 'react';

const MovieItemInfo = ({ movieTitle, releaseDate, genre }) => (
    <div className='item-info'>
        <div className='item-info-details'>
        <div className='film-title'>
            { movieTitle.length > 16 ? `${movieTitle.slice(0, 18)}...` : movieTitle }
        </div>
        <div className='release-date'>
            { releaseDate.slice(0, 4) }
        </div>
        </div>
        <div className='item-info-genre'>
            { genre.join(', ') }
        </div>
    </div>
);

export default MovieItemInfo;
