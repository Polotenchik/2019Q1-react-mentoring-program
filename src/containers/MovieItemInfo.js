import React from 'react';

export const MovieItemInfo = (props) => {

 // let title = filmTitle.length > 16 ? filmTitle.slice(0,18) + '...' :  filmTitle;

return (
  <div className='item-info'>
    <div className='item-info-details'>
      <div className='film-title'>
        { props.title }
      </div>
      <div className='release-date'>
        { props.releaseDate.slice(0,4) }
      </div>
    </div>
    <div className='item-info-genre'>
      { props.genre.join(', ') }
    </div>
  </div>
)
};
