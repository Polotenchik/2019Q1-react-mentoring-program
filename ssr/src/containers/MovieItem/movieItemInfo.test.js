import React from 'react';
import { MovieItemInfo } from './MovieItemInfo';
import { shallow } from 'enzyme';

const mockFilmTitleText = 'Star Wars';
const mockFilmReleaseDateText = '2017-08-08';
const mockFilmGenreArray = ['comedy', 'action', 'drama'];

const mockFilmTitleLongText = 'Guardians of the Galaxy. Episode 1. Part 2. Intro'

describe('<MovieItemInfo />', () => {

    it('should render the component', () => {
        const wrapper = shallow(
            <MovieItemInfo 
                movieTitle={ mockFilmTitleText }
                releaseDate={ mockFilmReleaseDateText }
                genre = { mockFilmGenreArray }
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should render the component with long title (more than 17 characters)', () => {
        const wrapper = shallow(
            <MovieItemInfo 
                movieTitle={ mockFilmTitleLongText }
                releaseDate={ mockFilmReleaseDateText }
                genre={ mockFilmGenreArray }
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});