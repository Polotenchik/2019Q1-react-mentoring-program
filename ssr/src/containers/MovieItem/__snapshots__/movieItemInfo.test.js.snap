// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieItemInfo /> should render the component 1`] = `
<div
  className="item-info"
>
  <div
    className="item-info-details"
  >
    <div
      className="film-title"
    >
      Star Wars
    </div>
    <div
      className="release-date"
    >
      2017
    </div>
  </div>
  <div
    className="item-info-genre"
  >
    comedy, action, drama
  </div>
</div>
`;

exports[`<MovieItemInfo /> should render the component with long title (more than 17 characters) 1`] = `
<div
  className="item-info"
>
  <div
    className="item-info-details"
  >
    <div
      className="film-title"
    >
      Guardians of the G...
    </div>
    <div
      className="release-date"
    >
      2017
    </div>
  </div>
  <div
    className="item-info-genre"
  >
    comedy, action, drama
  </div>
</div>
`;
